graphql-dgs-codegen-gradle/src/test/kotlin/com/netflix/graphql/dgs/CodegenGradlePluginSpringBootSmokeTest.kt

```diff
@@ -0,0 +1,277 @@
+/*
+ *
+ *  Copyright 2020 Netflix, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.netflix.graphql.dgs
+
+import org.assertj.core.api.Assertions.assertThat
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome.SUCCESS
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+
+class CodegenGradlePluginSpringBootSmokeTest {
+
+    @TempDir
+    lateinit var projectDir: File
+
+    val graphQLSchema =
+        """
+type Query {
+    result: Result!
+    find(filter: Filter!): Result!
+}
+
+type Result {
+    isSuccessful: Boolean
+    result: String
+}
+
+input Filter {
+    mandatoryString: String!
+    optionalString: String
+    mandatoryNumber: Int!
+    optionalNumber: Int
+}
+
+interface Audited {
+    lastUpdated: DateTime
+    lastUpdatedBy: String
+}
+
+interface MetaData implements Audited {
+    lastUpdated: DateTime
+    lastUpdatedBy: String
+}
+
+type JSONMetaData implements MetaData & Audited {
+    data: JSON
+    url: Url
+    lastUpdated: DateTime
+    lastUpdatedBy: String
+}
+
+type SimpleMetaData implements MetaData & Audited{
+    data: [String]
+    lastUpdated: DateTime
+    lastUpdatedBy: String
+}
+
+scalar Date 
+scalar DateTime
+scalar Time
+scalar JSON
+scalar Url
+            """.trimMargin()
+
+    @Test
+    fun `A SpringBoot project can use the generated Java`() {
+        prepareBuildGraphQLSchema(graphQLSchema)
+
+        prepareBuildGradleFile(
+            """
+plugins {
+    id 'java'
```

Comment:
So this will make the test use the local snapshot of the plugin?

Reply:
In this case the `GradleRunner` is the one that _injects_ the codegen-gradle code when it executes the build.
This test is executed as part of the normal _test_ task in Gradle.

In addition to this I'm looking at executing example projects as part of a CI chain, where the example projects run dev snapshots of the codegen plugin.